### ISSUE TITLE
segmentation fault error when use jsonpickle to serialize and de-serialize an object

### ISSUE THREAD

**Issue by helloyuepang** [jsonpickle/jsonpickle#439]
Hi, 
I tried to use jsonpickle to serialize an object. But after de-serialization, I met an error: segmentation fault. 

Following is my code:

```
import jsonpickle
import pandas as pd

class MyClass:
    def __init__(self, df):
        self.df = df

df = pd.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
my_class = MyClass(df)

json_string = jsonpickle.encode(my_class)

f = open('./test.json', 'w')
f.write(json_string)
f.close()

read_path = 'test.json'
f = open(read_path)
json_str = f.read()
obj = jsonpickle.decode(json_str)
print(obj.df)

```

And I found that if the data type of attribute in the class is list or int, instead of pandas dataframe, jsonpickle works. My python version is 3.8.8.

Is there any way to solve this problem?

Thanks!

**Comment by Theelx** [comment]
Hey there, sorry for the late response (midterms have been difficult lately). I believe you need to register the jsonpickle pandas handlers, there should be instructions on how to do that in the online documentation for jsonpickle. If that doesn't work, please let me know!

**Comment by Theelx** [comment]
Never mind, registering pandas handlers doesn't work. I'll try and figure this out, but it may be difficult.

**Comment by rishavganguly007** [comment]
> Never mind, registering pandas handlers doesn't work. I'll try and figure this out, but it may be difficult.

Hi,

I have run the code mentioned in the issue description with adding the pandas handlers and it worked fine for me.

Can you please provide more insights regarding this issue?

here's the code that I ran -> https://colab.research.google.com/drive/1il1cS8S-fFmT_bSyR0_buDfwrxxK6jvf?usp=sharing

Thanks

**Comment by Theelx** [comment]
Ah, the code I used did the following:
```py
import jsonpickle
import pandas as pd

jsonpickle.handlers.register('pandas')
```
instead of, the following, which works:
```py
import jsonpickle.ext.pandas as jsonpickle_pandas
jsonpickle_pandas.register_handlers()
import jsonpickle
import pandas as pd
```
Thanks for figuring this out @rishavganguly007, and thanks for testing it @helloyuepang! I'll try and figure out why the first method didn't work, and hopefully I can fix it for 3.0.2.
